platform/security/platform-security-core-api/src/main/java/ddf/security/principal/GuestPrincipal.java

```diff
@@ -65,9 +68,25 @@ public static String parseAddressFromName(String fullName) {
         if (!StringUtils.isEmpty(fullName)) {
             String[] parts = fullName.split(NAME_DELIMITER);
             if (parts.length == 2) {
-                return parts[1];
+                return formatIpAddress(parts[1]);
             }
         }
         return null;
     }
+
+    //IPv6 addresses should be contained within brackets to conform
+    //to the spec IETF RFC 2732
+    private static String formatIpAddress(String ipAddress) {
+        try {
+            if (InetAddress.getByName(ipAddress) instanceof Inet6Address) {
```

Comment:
Yes I could. Thinking about it there may be a simpler solution: 

```
InetAddress parsedIp = InetAddress.getByName(ipAddress);
if (parsedIp instanceof Inet6Address) {
   return  "[" + parsedIp.getHostAddress() + "]";
}
```

Since getHostAddress() returns only the IP Address wrapping it in brackets ensures it is always in the desired format.